src/java.base/share/classes/jdk/internal/math/FloatingDecimal.java

```diff
@@ -1872,7 +1872,7 @@ static ASCIIToBinaryConverter readJavaFormatString( String in ) throws NumberFor
                 }
             }  // look for and process decimal floating-point string
 
-            char[] digits = new char[ len ];
+            byte[] digits = new byte[ len ];
```

Comment:
```suggestion
            byte[] digits = new byte[len];
```

Reply:
@turbanoff I'll apply your suggestions later, in [PR 21840](https://github.com/openjdk/jdk/pull/21840)